tests/api/test_types.py

```diff
@@ -127,11 +158,49 @@ def test_sas_code():
             SASCode(invalid)
 
 
-def test_entryname_normalization():
-    nfc_str = "àáâäæãåāçćčèéêëēėęîïíīįìłñńôöòóœøōõßśšûüùúūÿžźż"
-    nfd_str = (
-        "àáâäæãåāçćčèéêëēėęîïíīįìłñńôöòóœøōõßśšûüùúūÿžźż"
-    )
+@pytest.mark.parametrize(
+    "data",
+    (
+        "foo",
+        "foo.txt",
+        "x" * 255,  # Max size
+        "飞" * 85,  # Unicode & max size
+        "X1-_é飞" "🌍☄️==🦕🦖💀",  # Probably a bad name for a real folder...
+        ".a",  # Dot and dot-dot are allowed if they are not alone
+        "..a",
+        "a..",
+        "a.",
+    ),
+)
+def test_valid_entry_name(data):
```

Comment:
How does this work ? Does the test pass as long as the `EntryName` call doesn't raise any errors ?